### PR TITLE
Prevent profile updates without changes

### LINKED ISSUE #2807: [Edit profile. Profile] The user be able to update information without any changes on the ‘Edit Profile’ page
**Environment:**

 Browser - Chrome 131.0.6778.70

**Reproducible:** always.

**Precondition:**

1. A user with the Student role and a user with the Tutor role are registered.
2. Log In as a :
    1)Tutor
    2)Student
3. On the 'Edit profile' page.

**Steps to reproduse:**

1. Navigate to Professional information =>Professional information=>Notifications=>Password & Security.
2. Reload the 'Edit profile' page without making any changes.
3. Click on the 'Profile' tab 

**Actual Result:**

The ‘Update’ button is available.

**Expected Result:**

The ‘Update’ button is disabled.

User story link:
User Story - #2693
TC – #2752

### PR DESCRIPTION
**Pull request by luiqor**
Fix:
The ‘Update’ button is disabled after refreshing the page:

https://github.com/user-attachments/assets/f6e4c93a-6754-4751-9f0e-bb7c3d55d2ac

The ‘Update’ button becomes active when there are new changes:
(all fields accurate work is demonstrated)

https://github.com/user-attachments/assets/116af3c0-644d-40a2-96b0-0c7b9c5259ac

Photo:
( Recordings are made before `Fixed changing a user photo in the header (#2812)` commit)
- The initial state and the current state of the photo are compared correctly.
- The ''expected by backend" data are sent to the server when the user removes the photo.

https://github.com/user-attachments/assets/612b7936-ccb8-40f3-b5ae-3a62574dcd07

https://github.com/user-attachments/assets/808ab86d-471a-453b-940b-cac2d9ad858e

VideoLink:

- The initial state and the current state of the videoLink are compared correctly.
- The videoLink type is represented accurately in the state.
- Only the videoLink associated with the current user role is sent to the server for updates.

https://github.com/user-attachments/assets/da887e97-4f89-4cef-949b-10fa40cfa201

Additional:

- Add fetching data from server before leaving the page to prevent showing not applied changes to user (For instance, avatar that was updated in redux state, but not updated on sever) [ As shown in the recording at 00:18 ](https://github.com/user-attachments/assets/3d8f4e5b-90a6-43de-9b32-385037685bcd)
- Extract separate utils, add tests for them

### CODE REVIEW

File: src/containers/edit-profile/profile-tab/ProfileTab.tsx [hunk 1]
```diff
@@ -62,7 +59,15 @@ const ProfileTab: FC = () => {
...
-    void dispatch(updateProfileData(data))
+    const payload = {
+      ...data,
+      videoLink: {
+        student: userRole === UserRoleEnum.Student ? data.videoLink : null,
+        tutor: userRole === UserRoleEnum.Tutor ? data.videoLink : null
+      }
+    }
+
+    void dispatch(updateProfileData(payload))
```
Comment (Made1ra) [inline src/containers/edit-profile/profile-tab/ProfileTab.tsx]: Consider removing the `void` operator for better maintainability.
[Link](https://sonarcloud.io/project/issues?id=ita-social-projects_SpaceToStudy-Client&pullRequest=2849&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true)

**Review by Made1ra — APPROVED** [review]

File: src/containers/edit-profile/profile-tab/ProfileTab.tsx (hunk 1 above)
Comment (luiqor) [reply src/containers/edit-profile/profile-tab/ProfileTab.tsx]: Resolved ✅

File: src/pages/edit-profile/EditProfile.tsx [hunk 2]
```diff
@@ -87,6 +88,10 @@ const EditProfile = () => {
       )
     }
     void fetchData()
+
+    return () => {
+      void fetchData()
+    }
```
Comment (ShadowOfTheSpace) [inline src/pages/edit-profile/EditProfile.tsx:94]: Why `fetchData` is using as cleanup function?

Comment (luiqor) [reply src/pages/edit-profile/EditProfile.tsx:94]: We have a logic where the photo updates immediately when the user selects a new one (either by clicking the "Select" or "Remove Photo" button), even before clicking the "Update" button. However, when the user leaves the edit page, the photo that hasn’t been updated (saved to the server) is still shown as their profile icon in the top-right corner.
To address this, before the component is unmounted and the user leaves the edit profile page, I am refetching the data from the server to ensure that only the correctly updated photo is displayed.

Comment (ShadowOfTheSpace) [reply src/pages/edit-profile/EditProfile.tsx:94]: Oh, I got it
It's very thoughtfully, really)

**Review by ShadowOfTheSpace — APPROVED** [review]

**Review by amoutens — APPROVED** [review]

**Pull request merged**